### PR TITLE
C# Ctor ApiCall partial methods

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

File: src/main/resources/com/google/api/codegen/csharp/gapic_client.snip [hunk 1]
```diff
@@ -403,9 +403,42 @@
                     *** ERROR: Cannot handle grpc streaming type '{@call.grpcStreamingType.toString}' ***
                 @end
             @end
+            @join call : api.apiCallableMembers
+                # call: ApiCallableView
+                OnConstruction__ApiCall(ref {@call.name});
```
[the remark this answers is not in the document]
Comment (chrisdunelm) [reply src/main/resources/com/google/api/codegen/csharp/gapic_client.snip]: The `Modify_` names are for methods called during a single RPC. I'd like to not have a call from the constructor have the same naming convention. Very happy to choose something that is better than `OnConstruction_...`, although `OnConstruction_...` does make it clear its similar to the existing `OnConstruction` method.

The double underscore is delibrate. If the service has an RPC method called `Api` (OK, unlikely) then having just a single underscore will cause a name collision. Again, very happy to change that name to something better.

**Comment by chrisdunelm** [comment]
@jskeet PTAL

**Comment by chrisdunelm** [comment]
I like the `Modify_FooApiCall` naming, I'll change it again...

File: src/test/java/com/google/api/codegen/testdata/csharp/csharp_library.baseline [hunk 2]
```diff
@@ -10251,62 +10251,66 @@ namespace Google.Example.Library.V1
...
+            Modify_GetBigNothingApiCall(ref _callGetBigNothing);
             Modify_ApiCall(ref _callTestOptionalRequiredFlatteningParams);
-            Modify_ApiCall_TestOptionalRequiredFlatteningParams(ref _callTestOptionalRequiredFlatteningParams);
+            Modify_TestOptionalRequiredFlatteningParamsApiCall(ref _callTestOptionalRequiredFlatteningParams);
             OnConstruction(grpcClient, effectiveSettings, clientHelper);
         }
 
+        // Partial methods are named to (mostly) ensure there cannot be conflicts with RPC method names.
+
+        // Partial methods called for every ApiCall on construction.
```
Comment (jskeet) [inline src/test/java/com/google/api/codegen/testdata/csharp/csharp_library.baseline]: Nit: *maybe* we should specify that the generic call is made before the specific one?
(I'm not overly keen on "modification of all the underlying ApiCall objects" given that they're immutable, but it makes sense with the Modify method name, and I suspect it'll be clear enough to everyone actually using this facility.)

**Comment by chrisdunelm** [comment]
Merged. We can update the comment later ;)

**Pull request merged**